.github/workflows/new_linter_checklist.yaml

```diff
@@ -0,0 +1,105 @@
+---
+name: Add new linter checklist
+on:
+  pull_request:
+    types:
+      - labeled
+
+jobs:
+  add-comment:
+    if: "github.event.label.name == 'linter: new'"
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+
+    steps:
+      - name: Add checklist
+        run: |
+          # This is just safety to not spam with new comments if the tag is
+          # removed and added back multiple times maliciously. It will reset the
+          # checklist but the previous values will be in the edit history.
+          comment_exist=$(gh pr view "$NUMBER" \
+            --json comments \
+            --jq '.comments[].author | select(.login=="github-actions") | .login' \
+            | wc -l)
+          [ "$comment_exist" -gt 0 ] && edit_last="--edit-last"
+
+          # Comment on the PR with the checklist.
+          gh pr comment "$NUMBER" --body "$BODY" $edit_last
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GH_REPO: ${{ github.repository }}
+          NUMBER: ${{ github.event.number }}
+          BODY: |
```

Comment:
I think it's better to use a dedicated markdown file instead of a multiple YAML string inside an env var.